Khan/perseus
# Remove ItemRenderer again, but make ServerItemRenderer work with linting

## Summary:
This reverts  #847 which reverted  #833. It also makes ServerItemRenderer work with `linterContext` so it can be used in the content editor.

Issue: LC-1546

## Test plan:
Once this lands to Webapp:
- Go to the content editor
- Go to some page where you can edit content
- Make a linting error (like writing `# hello world\n\n`)
- Note that ServerItemRenderer shows a linting error
